R/poisdm.R

```diff
@@ -1,103 +1,79 @@
-#' Poisson DataMaker
+#' Poisson datamaker
 #'
-#' @param n Sample size
-#' @param lambda Mean of sample
-#' @param groupnames Names of groups
+#' @param n sample size
+#' @param lambda group mean
+#' @param groupnames names of the groups
 #'
-#' @return Table with input parameters, histogram of simulated values, and scatterpoint of simulated values
+#' @return
 #' @export
+#' @importFrom rlang .data
 #'
 #' @examples
 #' poisdm(c(40, 50), c(10, 12), c("group1", "group2"))
-#' poisdm(c(40, 50), c(10, 12), c("group2"))
-#' poisdm(c(40, 50), c(-10, 12), c("group1", "group2"))
-#' poisdm(rep(40, 5), rep(10, 5), c("group", "group2", "group3", "group4", "group5"
 
 poisdm <- function(n, lambda, groupnames){
-
-  # make data frame of input values
   input <- data.frame(
     groupnames
     , n
     , lambda
   )
 
-  # error for unequal input lengths
-  if(length(n) != length(lambda) || length(n) != length(groupnames)){
+  if(length(n) != length(lambda) | length(n) != length(groupnames)){
     print("specify an equal number of sample sizes (n), means (lambda), and group names")
     stop()
   }
 
-  # error for negative input
-  if (!all((lambda >= 0) && (n >= 0))) {
+  if (!all((lambda >= 0) & (n >= 0))) {
     print("lambda (vector of means) and n (sample size) must be non-negative")
     stop()
   }
 
-  # generate random samples - list of length(n), int[1:n[i]]
   df1 <- lapply(1:length(n)
-                , FUN = function(i){rpois(n = n[i]
+                , FUN = function(i){stats::rpois(n = n[i]
                                           , lambda = lambda[i]
                 )
                 }
   )
 
-  # makes data frame with length(n) obs of max(n) variables
-  # fills difference in output lengths with NAs
   df2 <- purrr::map_dfr(
     .x = df1
     , .f = ~ as.data.frame(
       t(.)
     )
   )
 
-  # make it a data frame and flips
   df2 <- as.data.frame(t(df2))
 
-  # rename columns with group names
   colnames(df2) <- groupnames
 
-  # prep data to plot (make it tidy!)
   dfp <- stats::na.omit(tidyr::pivot_longer(data = df2
```

Comment:
this is the object that the repeater function will conduct the statistical test upon...so it is the most important object of all!!  see jabstb 37.3.4

so when a use runs poisdm, this object is the default output